app/scripts/metamask-controller.js

```diff
@@ -188,129 +190,31 @@ export default class MetamaskController extends EventEmitter {
       this.accountTracker._updateAccounts()
     })
 
-    const additionalKeyrings = [TrezorKeyring, LedgerBridgeKeyring]
-    this.keyringController = new KeyringController({
-      keyringTypes: additionalKeyrings,
-      initState: initState.KeyringController,
-      getNetwork: this.networkController.getNetworkState.bind(this.networkController),
-      encryptor: opts.encryptor || undefined,
-    })
-    this.keyringController.memStore.subscribe((s) => this._onKeyringControllerUpdate(s))
-    this.keyringController.on('unlock', () => this.emit('unlock'))
-
-    this.permissionsController = new PermissionsController({
```

Comment:
note to reviewers: any controllers which depend on `this.keyringController` were moved into a separate function so they could be re-initialized when `this.keyringController` changes to a different KeyringController. the ones that didn't seem to depend on `this.keyringController` were left as-is.